src/main/java/org/opentripplanner/routing/edgetype/SemiPermanentPartialStreetEdge.java

```diff
@@ -0,0 +1,88 @@
+package org.opentripplanner.routing.edgetype;
+
+import com.vividsolutions.jts.geom.LineString;
+import org.opentripplanner.common.model.P2;
+import org.opentripplanner.routing.core.RoutingRequest;
+import org.opentripplanner.routing.core.State;
+import org.opentripplanner.routing.vertextype.SplitterVertex;
+import org.opentripplanner.routing.vertextype.StreetVertex;
+import org.opentripplanner.routing.vertextype.TemporarySplitterVertex;
+import org.opentripplanner.util.I18NString;
+
+/**
+ * This class is used to model a SemiPermanent StreetEdge that was non-destructively split from another StreetEdge.
+ * These StreetEdges are typically used to model access to things such as bike rental stations that are dynamically
+ * created from bike rental updaters and could be subsequently removed from the feed. These edges aid in providing
+ * access only when needed to the desired bike rental stations and also allow for the isolated removal of these edges
+ * when they are no longer needed.
```

Comment:
This javadoc might also mention the concept of floating vehicles, which might help clarify the purpose of this class a little better.

Reply:
I'll reference some GitHub issues, but perhaps in the distant future, there may be other uses for creating such an edge type. Like maybe these edge types could model access to teleportation portals that are created for a few hours with a connection to a specific street edge.